rdtools/system_analysis.py

```diff
@@ -0,0 +1,585 @@
+'''
```

Comment:
Even if many of the underlying components are already covered by unit tests, the basic functionality of the analysis class in this module should also be covered by unit tests. I don't think it's necessary to try for complete coverage right away. Perhaps all that is needed for the near term is a simple check that this module can be imported and the main class can be instantiated.